tests/test_sbi.py

```diff
@@ -137,100 +206,142 @@ def test_snle(monkeypatch):
-        posterior=posterior,
-        x_obs=x[ind], theta_fid=theta[ind],
-        x=x, theta=theta
-    )
+        # generate samples from the posterior using MCMC
+        samples = posterior.sample(
+            (nsamples,), x[ind],
+            method='slice_np_vectorized', num_chains=2
+        ).detach().cpu().numpy()
+
+        # calculate the potential (prop. to log_prob) for each sample
+        log_prob = posterior.log_prob(
+            nsamples,
+            x[ind]
+        ).detach().cpu().numpy()
+
+        # use ltu-ili's built-in validation metrics to plot the posterior
+        metric = PlotSinglePosterior(
+            backend='sbi', output_path=Path('./toy'), num_samples=nsamples,
+            sample_method='slice_np_vectorized',
+            sample_params={'num_chains': 2, 'burn_in': 1, 'thin': 1},
+            labels=[f'$\\theta_{i}$' for i in range(npar)],
+            seed=1, save_samples=True,
+        )
+        fig = metric(
+            posterior=posterior,
+            x_obs=x[ind], theta_fid=theta[ind],
+            x=x, theta=theta
+        )
 
-    metric = PlotSinglePosterior(
-        backend='sbi', output_path=None, num_samples=nsamples,
-        sample_method='vi',
-        sample_params={'dist': 'maf',
-                       'n_particles': 32, 'learning_rate': 0.01},
-        labels=[f'$\\theta_{i}$' for i in range(3)]
-    )
-    fig = metric(
-        posterior=posterior,
-        x_obs=x[ind], theta_fid=theta[ind],
-        x=x, theta=theta
-    )
+        metric = PlotSinglePosterior(
+            backend='sbi', output_path=None, num_samples=nsamples,
+            sample_method='vi',
+            sample_params={'dist': 'maf',
+                           'n_particles': 32, 'learning_rate': 0.01},
+            labels=[f'$\\theta_{i}$' for i in range(npar)]
+        )
+        fig = metric(
+            posterior=posterior,
+            x_obs=x[ind], theta_fid=theta[ind],
+            x=x, theta=theta
+        )
+        
+        if npar == 1:
+            # Cannot sample directly for snle
+            metric = PosteriorCoverage(
+                backend='sbi', output_path=None, num_samples=nsamples,
+                sample_method='direct', labels=[f'$\\theta_{i}$' for i in range(npar)],
+                plot_list=["predictions", "coverage", "histogram"]
+            )
+            unittest.TestCase().assertRaises(
+                ValueError,
+                metric,
+                posterior=posterior,
+                x_obs=x[ind], 
+                theta_fid=theta[ind],
+                x=x[:2], 
+                theta=theta[:2]
+            )
+            
+            # Can sample with vi for snle
+            metric = PosteriorCoverage(
+                backend='sbi', output_path=None, num_samples=nsamples,
+                sample_method='vi', labels=[f'$\\theta_{i}$' for i in range(npar)],
+                plot_list=["predictions", "coverage", "histogram"]
+            )
+#             metric(
```

Comment:
Remove?

Reply:
I've uncommented this. I thought that this test was extremely slow, but it only takes 30 seconds when run locally, so I have uncommented.